app/models/katello/glue/candlepin/pool.rb

```diff
@@ -145,14 +145,21 @@ def import_data(index_hosts = true)
       def import_hosts
         entitlements = Resources::Candlepin::Pool.entitlements(self.cp_id, ["consumer.uuid"])
         uuids = entitlements.map { |ent| ent["consumer"]["uuid"] }
-        sub_facets = Katello::Host::SubscriptionFacet.where(:uuid => uuids)
-        sub_facet_ids = sub_facets.pluck(:id)
-        sub_facet_ids.each do |sub_facet_id|
-          Katello::SubscriptionFacetPool.where(:pool_id => self.id, :subscription_facet_id => sub_facet_id).first_or_create
-        end
-        existing_associations = Katello::SubscriptionFacetPool.where(:pool_id => self.id)
-        correct_associations = Katello::SubscriptionFacetPool.where(:pool_id => self.id, :subscription_facet_id => sub_facet_ids)
-        (existing_associations - correct_associations).map(&:destroy)
+
+        sub_facet_ids_from_cp = Katello::Host::SubscriptionFacet.where(:uuid => uuids).select(:id).pluck(:id)
+        sub_facet_ids_from_pool_table = Katello::SubscriptionFacetPool.where(:pool_id => self.id).select(:subscription_facet_id).pluck(:subscription_facet_id)
+        entries_to_add = sub_facet_ids_from_cp - sub_facet_ids_from_pool_table
+        unless entries_to_add.empty?
+          ActiveRecord::Base.transaction do
```

Comment:
why put it in a transaction?

Reply:
hmm I followed the pattern here -> https://github.com/Katello/katello/blob/master/app/models/katello/concerns/pulp_database_unit.rb#L131-L135

Assumed it might be faster to group  a bunch of inserts in a single transaction. I thought of using single insert statements but I am kinda averse to have a single insert statement insert upto 50K entries.

Reply:
+1